lib/auth/native/native_test.go

```diff
@@ -203,6 +203,18 @@ func (s *NativeSuite) TestUserCertCompatibility(c *check.C) {
 	caSigner, err := ssh.ParsePrivateKey(priv)
 	c.Assert(err, check.IsNil)
 
+	roleFoo, err := types.NewRole("foo", types.RoleSpecV4{
```

Comment:
add validation to the `types.CertExtension`  CertExtensions type
For instance the `types.NewRole` call for   RoleSpecV4 with `CertExtension.Type: "invalid type ",`  should return an error.